app/lib/frontend/handlers/package.dart

```diff
@@ -324,61 +325,72 @@ Future<shelf.Response> packageActivityLogHandler(
 
 @visibleForTesting
 Future<PackagePageData> loadPackagePageData(
-  String packageName,
+  Package package,
   String? versionName,
   String? assetKind,
 ) async {
-  final package = await packageBackend.lookupPackage(packageName);
-  if (package == null || package.isNotVisible) {
-    // note: this should not happen, as isVisible check is done before calling the method
-    throw NotFoundException.resource('package "$package"');
-  }
+  final packageName = package.name!;
+  versionName ??= package.latestVersion;
 
-  final bool isLiked = requestContext.isNotAuthenticated
-      ? false
-      : await likeBackend.getPackageLikeStatus(
-              requestContext.authenticatedUserId!, package.name!) !=
-          null;
+  final latestReleasesFuture = packageBackend.latestReleases(package);
 
-  versionName ??= package.latestVersion;
-  final selectedVersion =
-      await packageBackend.lookupPackageVersion(packageName, versionName!);
+  final isLikedFuture = Future(
+    () async => requestContext.isNotAuthenticated
+        ? false
+        : await likeBackend.getPackageLikeStatus(
+                requestContext.authenticatedUserId!, package.name!) !=
+            null,
+  );
+
+  final selectedVersionFuture =
+      packageBackend.lookupPackageVersion(packageName, versionName!);
+  final versionInfoFuture =
+      packageBackend.lookupPackageVersionInfo(packageName, versionName);
+
+  final assetFuture = Future(() async => assetKind == null
+      ? null
+      : packageBackend.lookupPackageVersionAsset(
+          packageName, versionName!, assetKind));
+
+  final isAdminFuture = Future(() async => requestContext.isNotAuthenticated
+      ? false
+      : await packageBackend.isPackageAdmin(
+          package, requestContext.authenticatedUserId!));
```

Comment:
nit: How about?
```suggestion
  final isAdminFuture = requestContext.isNotAuthenticated
      ? Future.value(false)
      : await packageBackend.isPackageAdmin(
          package,
          requestContext.authenticatedUserId!,
        );
```

trailing comma, and using `Future.value(false)` is a tiny bit easier to read (perhaps).

Reply:
updated this and the above suggestions as a separate commit, as we had merge conflicts